eth/backend.go

```diff
@@ -1046,7 +1046,7 @@ func (s *Ethereum) Init(stack *node.Node, config *ethconfig.Config, chainConfig
 		}
 	}
 
-	s.apiList = jsonrpc.APIList(chainKv, ethRpcClient, txPoolRpcClient, miningRpcClient, ff, stateCache, blockReader, &httpRpcCfg, s.engine, s.logger)
+	s.apiList = jsonrpc.APIList(chainKv, ethRpcClient, txPoolRpcClient, miningRpcClient, ff, stateCache, blockReader, &httpRpcCfg, s.engine, s.logger, s.engine.(*bor.Bor).BridgeReader)
```

Comment:
Need to check if casting is possible and for nil BridgeReader.

Reply:
Compilation would fail if casting isn’t possible. nil checks are performed when the bridge reader is used.

Reply:
Yes, casting works but is not great for this purpose imho. @shohamc1 I think we've talked about this last week but maybe it got missed. Ideally, we do not use the `Bor` struct as a "provider" of the "BridgeReader", but instead simply initialise the "BridgeReader" (based on the config feature flag that tells us whether to use it or not) and attach it to the `Ethereum` backend struct as part of `New(...)` in backend.go so that it can be accessed later in `Init`. This is analogous to how we attached `polygonSyncService` to the backend struct too.